src/core/qgsrelation.cpp

```diff
@@ -187,18 +187,19 @@ QgsFeatureRequest QgsRelation::getReferencedFeatureRequest( const QgsAttributes&
   {
     int referencedIdx = referencedLayer()->fields().indexFromName( fieldPair.referencedField() );
     int referencingIdx = referencingLayer()->fields().indexFromName( fieldPair.referencingField() );
+    Q_UNUSED( referencingIdx );
 
     QgsField referencedField = referencedLayer()->fields().at( referencedIdx );
 
     if ( referencedField.type() == QVariant::String )
     {
       // Use quotes
-      conditions << QString( "\"%1\" = '%2'" ).arg( fieldPair.referencedField(), attributes.at( referencingIdx ).toString() );
+      conditions << QString( "\"%1\" = '%2'" ).arg( fieldPair.referencedField(), attributes.at( referencedIdx ).toString() );
```

Comment:
Not in this bug use case. But I'll check the caller to see why the attributes are coming from the wrong side and if is there any alternate function to call instead.